hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -650,6 +654,8 @@ private void handleFinishedAndStoreInTargetStatus(final JpaAction action) {
             target.setUpdateStatus(TargetUpdateStatus.IN_SYNC);
         }
 
+        targetManagement.requestControllerAttributes(target.getControllerId());
```

Comment:
Isn't it better to first save the `target` and if this is successfully ask for controller attributes? Otherwise it may happen that you request attributes while update is still pending